src/librustc/hir/print.rs

```diff
@@ -1314,7 +1314,12 @@ impl<'a> State<'a> {
                 }
                 self.bclose_(expr.span, indent_unit)?;
             }
-            hir::ExprClosure(capture_clause, ref decl, body, _fn_decl_span) => {
+            hir::ExprClosure(capture_clause, ref decl, body, _fn_decl_span, gen) => {
+                if gen.is_some() {
+                    self.head("gen")?;
+                    space(&mut self.s)?;
+                }
```

Comment:
We need syntax to mark a generator as movable or immovable. I haven't come up with any good ideas for that. The best idea I got is using `static` for immovable generators. The only syntax that seems to be nailed down is `yield`